src/selection/selectChild.js

```diff
@@ -0,0 +1,17 @@
+import matcher from "../matcher.js";
+import constant from "../constant.js";
+
+var find = Array.prototype.find;
+
+function childFind(match) {
+  match = typeof match === "function" ? match
+        : match == null ? constant(true)
+        : matcher(match);
+  return function() {
+    return find.call(this.children, function(e) { return match.call(e); });
+  };
+}
+
+export default function(match) {
+  return this.select(childFind(match));
```

Comment:
For the null case, this could be optimized by avoiding find entirely. And we can move the type checking of *match* to this function rather than in childFind, like you do for selectChildren.

```suggestion
  return this.select(match == null ? childFirst
      : childFind(typeof match === "function" ? match : childMatcher(match)));
```

where

```js
function childFirst() {
  return this.firstElementChild;
}
```